compiler/tests-full/stdlib.cma.expected.js

```diff
@@ -15075,85 +15078,93 @@
              /*<<camlinternalFormat.ml:817:38>>*/ return [11,
                     trans(rest1$10, rest2$10)];
-           default: switch$11 = 1;
+           default: switch$12 = 1;
          }
-        if(switch$11)
+        if(switch$12)
           /*<<camlinternalFormat.ml:818:21>>*/ throw  /*<<camlinternalFormat.ml:818:21>>*/ caml_maybe_attach_backtrace
                 ([0, Assert_failure, _n_], 1);
         break;
        case 12:
-        var rest1$11 = ty1[1], switch$12 = 0;
+        var rest1$11 = ty1[1], switch$13 = 0;
         if(typeof ty2 === "number")
-         switch$12 = 1;
+         switch$13 = 1;
         else
          switch(ty2[0]){
-           case 10: break;
+           case 10:
+            switch$1 = 1; break;
            case 11:
-            switch$0 = 1; break;
+            switch$0 = 1; switch$1 = 1; break;
            case 12:
             var rest2$11 = ty2[1];
              /*<<camlinternalFormat.ml:821:34>>*/ return [12,
                     trans(rest1$11, rest2$11)];
-           default: switch$12 = 1;
+           default: switch$13 = 1;
          }
-        if(switch$12)
+        if(switch$13)
           /*<<camlinternalFormat.ml:822:19>>*/ throw  /*<<camlinternalFormat.ml:822:19>>*/ caml_maybe_attach_backtrace
                 ([0, Assert_failure, _o_], 1);
         break;
        case 13:
-        var rest1$12 = ty1[1], switch$13 = 0;
+        var rest1$12 = ty1[1], switch$14 = 0;
         if(typeof ty2 === "number")
-         switch$13 = 1;
+         switch$14 = 1;
         else
          switch(ty2[0]){
-           case 10: break;
+           case 10:
+            switch$1 = 1; break;
            case 11:
-            switch$0 = 1; break;
+            switch$0 = 1; switch$1 = 1; break;
            case 12:
-            switch$0 = 2; break;
+            switch$0 = 2; switch$1 = 1; break;
            case 13:
             var rest2$12 = ty2[1];
              /*<<camlinternalFormat.ml:825:40>>*/ return [13,
                     trans(rest1$12, rest2$12)];
-           default: switch$13 = 1;
+           default: switch$14 = 1;
          }
-        if(switch$13)
+        if(switch$14)
           /*<<camlinternalFormat.ml:826:22>>*/ throw  /*<<camlinternalFormat.ml:826:22>>*/ caml_maybe_attach_backtrace
                 ([0, Assert_failure, _p_], 1);
         break;
        default:
-        var rest1$13 = ty1[1], switch$14 = 0;
+        var rest1$13 = ty1[1], switch$15 = 0;
         if(typeof ty2 === "number")
-         switch$14 = 1;
+         switch$15 = 1;
         else
          switch(ty2[0]){
-           case 10: break;
+           case 10:
+            switch$1 = 1; break;
            case 11:
-            switch$0 = 1; break;
+            switch$0 = 1; switch$1 = 1; break;
            case 12:
-            switch$0 = 2; break;
+            switch$0 = 2; switch$1 = 1; break;
            case 13:
-            switch$0 = 3; break;
+            switch$0 = 3; switch$1 = 1; break;
```

Comment:
The diff in this file may look like a regression (in term of code size) but it looks much better once rebased on top of #1496